### PR TITLE
TINY-4113: Now saves both sources if only one previously existed.

### PR DESCRIPTION
**Pull request by HAFRMO**
Related Ticket: TINY-4113

Description of Changes:
* Placeholder text

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] License headers added on new files (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/plugins/media/main/ts/core/UpdateHtml.ts [hunk 1]
```diff
@@ -161,7 +161,7 @@ const updateHtml = (html: string, data: Partial<MediaData>, updateAll?: boolean)
               const attrs: any = [];
               attrs.map = {};
 
-              if (sourceCount < index) {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/media/main/ts/core/UpdateHtml.ts:164]: This seems like the correct fix though and it seems to have broken in https://github.com/tinymce/tinymce/commit/1fc08c7ecdc2faab9d3a299be3297c907e1375ed when the offsets where changed.

File: modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import * as UpdateHtml from '../../../main/ts/core/UpdateHtml';
+import { Assert, UnitTest } from '@ephox/bedrock-client';
+import { MediaData } from 'tinymce/plugins/media/core/Types';
+
+UnitTest.test('atomic.core.HtmlToDataTest', () => {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts:5]: This name is almost definitely wrong. It should end with `UpdateHtmlTest` and the prefix looks incorrect as well. I think this should be correct:
```
atomic.plugins.media.UpdateHtmlTest
```

File: modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import * as UpdateHtml from '../../../main/ts/core/UpdateHtml';
```
Comment (lnewson) [inline modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts:1]: Import path is incorrect here. It should be `tinymce/plugins/media/core/UpdateHtml`

File: modules/tinymce/changelog.txt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Version 5.5.0 (TBD)
-
+    Fixed `media` not saving alternative source url in some situations.
```
Comment (lnewson) [inline modules/tinymce/changelog.txt:2]: There's a few minor grammar issues here, there should be no fullstop at the end and the JIRA number is missing. So it should probably be something like this:
```
Fixed the `media` plugin not saving the alternative source url in some situations #TINY-4113
```

File: modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts (hunk 3 above)
Comment (HAFRMO) [reply modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts:1]: When do we want relative paths and when do we want the `tinymce/...` types? I'm not feeling quite clear on it.

File: modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts (hunk 2 above)
Comment (HAFRMO) [reply modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts:5]: That'd likely be correct.

File: modules/tinymce/changelog.txt (hunk 4 above)
Comment (HAFRMO) [reply modules/tinymce/changelog.txt:2]: Swing and a miss. Better luck next time.

File: modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts (hunk 3 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts]: The rule is if it's in the same directory (eg `main` or `test) it should be relative. If it's outside of that then it should be an alias.

Comment (HAFRMO) [reply modules/tinymce/src/plugins/media/test/ts/atomic/UpdateHtmlTest.ts]: Makes sense. I'll keep to it.

**Pull request merged**